# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### Freestyle - Growth/Recovery/Momentum Prediction

Growth/Recovery/Momentum Prediction is the lane I have found the most interesting. Unlike the other lanes that describe existing data, this lane offers future insight that could prove valuable and more difficult to implement. I belief this lane will be the lane that would challenge me the most across the duration of the project, providing me with the most opportunity to grow my skill tree further.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

I believe this question improves the content team's prioritisation. Estimation of future page response could provide writers with a guide on which page would have a declining response, and therefore make updates to recover said page. 
A wrong recommendation cost would be the cost of effort from the writer, wherein they update a page that could recover on its own, in the case of a false positive and ignorance of a declining page would cause the page to be harder to recover with a lost of engagement in the case of the false negative. 

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [9]:
import pandas as pd
import numpy as np

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

# 1. how many pages are declining vs growin
print(df["trend_direction"].value_counts(normalize=True))

active = df[df["impressions_90d"] > 100]
print(active["trend_direction"].value_counts(normalize=True))

# 3. avg position by trend direction
print(df.groupby("trend_direction")["avg_position"].mean())

print(df["content_age_days"].std())


trend_direction
down      0.542067
stable    0.198733
up        0.146267
new       0.074533
flat      0.038400
Name: proportion, dtype: float64
trend_direction
down      0.597708
stable    0.237747
up        0.153360
new       0.009548
flat      0.001637
Name: proportion, dtype: float64
trend_direction
down      15.936305
flat      11.102431
new        9.912433
stable    16.332623
up        22.512739
Name: avg_position, dtype: float64
132.70793001795204


These numbers show that around 59.8% of active pages are declining, which is rise from the 54.2% decline population whilst comparing inactive pages, ensuring that the number of declining pages are not inflated by inactive or archived pages.
One of the factors that could affect page impressions, average position, shows that on average declining pages rank better than recovering pages (15.9 < 22.5) thus showing that pages that have good positions might not have good impressions, therefore simple rule of thumbs will not give good estimations. Hence, a model is needed to produce a better list for update prioritisation.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What I can claim?
* I can claim that some observed signals have causation to whether pages remains stable, or decline.
* I can claim that pages flagged as at risk to decline have more chances than pages that are not flagged at risk.
* I can claim that the ranked list provided by the model would serve better than random prioritisation made by the content team.

### What I cannot claim?
* I cannot claim that I am predicting the Google algorithm or pagerank system.
* I cannot claim that my model judges content quality, it only measures response to pages
* I cannot claim that a page flagged as stable, or increasing will not decline.
* I cannot claim that a refresh could cause a page to recover

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.